spx-gui/src/utils/audio.ts

```diff
@@ -101,6 +101,107 @@ export const useAudioDuration = (audio: () => string | Blob | null) => {
         return ''
       }
       return formatDuration(duration.value)
+    }),
+    formattedDurationSeconds: computed(() => {
+      if (duration.value === null || duration.value === Infinity) {
+        return ''
+      }
+      return duration.value.toFixed(1) + 's'
     })
   }
 }
+
+export async function trimAndApplyGain(
+  webmBlob: Blob,
+  startRatio: number,
+  endRatio: number,
+  gainValue: number
+): Promise<Blob> {
+  // Ensure ratios are between 0.0 and 1.0
+  if (
+    startRatio < 0.0 ||
+    startRatio > 1.0 ||
+    endRatio < 0.0 ||
+    endRatio > 1.0 ||
+    startRatio >= endRatio
+  ) {
+    throw new Error('Invalid start or end ratio')
+  }
+
+  const audioContext = new AudioContext()
```

Comment:
虽然没去细究原因，不过之前有看到 audio context 是建议单例复用的，详见这个文件中的 `getAudioContext`，这边可以也注意下

另外，这里是不是也应该用 `OfflineAudioContext`？

Reply:
为什么说 OfflineAudioContext 会更合适？官方的示例里是用的 AudioContext 解码再用 Offline 渲染

Reply:
我粗略看了下 OfflineAudioContext 的说明

> In contrast with a standard AudioContext, an OfflineAudioContext doesn't render the audio to the device hardware; instead, it generates it, as fast as it can, and outputs the result to an AudioBuffer.

只从这个描述出发我们的场景好像适合 OfflineAudioContext；不过我没深入看，有可能理解得不对

Reply:
我们在 trim 的时候是直接对 `AudioBuffer` 进行操作，应该是不需要由 `OfflineAudioContext` 来渲染的（wav也是一样），只在 gain 的时候需要重新渲染音频

Reply:
嗯只说了影响“render”行为；对于非 render 行为，预期 `OfflineAudioContext` 跟 `AudioContext` 是等价的对吧？